### PR TITLE
Add Test coverage section at developer-tools page

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
Screen Shot 2022-01-26 at 10 52 10 AM

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
This is a draft which I'm working on now in order to give a summary of our infra.

**Comment by dongjoon-hyun** [comment]
Hi, @HyukjinKwon . Could you add some about the additional Python coverage tools which we are currently using?

**Comment by srowen** [comment]
Just generate the html when done, yes. Looks ok

**Comment by dongjoon-hyun** [comment]
Thank you, @srowen . Sure!

**Review by viirya — COMMENTED** [review]
Looks good.

File: developer-tools.md [hunk 1]
```diff
@@ -7,6 +7,30 @@ navigation:
...
+- TPCDS ScaleFactor=1 Benchmark
+- JDBC docker tests
+- Java/Scala/Python/R unit tests with Java 8
+
+<h3 id="github_action">[AppVeyor](https://ci.appveyor.com/project/ApacheSoftwareFoundation/spark)</h3>
+
+AppVeyor provides the following on Windows.
+- Java/Scala/R unit tests with Java 8/Scala 2.12/SBT
+
+<h3 id="github_action">[Scaleway](https://www.scaleway.com)</h3>
```
Comment (viirya) [inline developer-tools.md]: The html `id` is correct?

Comment (dongjoon-hyun) [reply developer-tools.md]: Oops. Thank you!

**Comment by dongjoon-hyun** [comment]
Hi, All. I addressed the review comments and attached the generate HTML image into the PR description.

File: developer-tools.md [hunk 2]
```diff
@@ -7,6 +7,31 @@ navigation:
...
+
+Apache Spark community uses various resources to maintain the community test coverage.
+
+<h3 id="github_action">GitHub Action</h3>
+
+[GitHub Action](https://github.com/apache/spark/actions) provides the following on Ubuntu 20.04.
+- Scala 2.12/2.13 SBT build with Java 8
+- Scala 2.12 Maven build with Java 11/17
+- Java/Scala/Python/R unit tests with Java 8/Scala 2.12/SBT
+- TPCDS ScaleFactor=1 Benchmark
```
Comment (HyukjinKwon) [inline developer-tools.md]: Should we maybe mention about Java version here too? It's a nit but I would call it like `TPC-DS benchmark with scale factor 1`

File: developer-tools.md [hunk 3]
```diff
@@ -7,6 +7,31 @@ navigation:
...
+- JDBC docker integration tests
```
Comment (HyukjinKwon) [inline developer-tools.md]:
```suggestion
- JDBC Docker integration tests
```

File: developer-tools.md [hunk 4]
```diff
@@ -7,6 +7,31 @@ navigation:
...
+- Daily Java/Scala/Python/R unit tests with Java 11/17 and Scala 2.12/SBT
+
+<h3 id="appveyor">AppVeyor</h3>
+
+[AppVeyor](https://ci.appveyor.com/project/ApacheSoftwareFoundation/spark) provides the following on Windows.
+- Java/Scala/R unit tests with Java 8/Scala 2.12/SBT
```
Comment (HyukjinKwon) [inline developer-tools.md]: I believe this runs only R unit tests with Java 8/Scala 2.12/SBT

File: developer-tools.md [hunk 5]
```diff
@@ -7,6 +7,31 @@ navigation:
...
+
+<h3 id="scaleway">Scaleway</h3>
+
+[Scaleway](https://www.scaleway.com) provides the following on MacOS and Apple Silicon.
```
Comment (HyukjinKwon) [inline developer-tools.md:31]: This is awesome!

**Review by HyukjinKwon — COMMENTED** [review]
I think we don't have to mention about PySpark coverage stuff .. since that doesn't actually cover something new in terms of the test coverage.

**Comment by HyukjinKwon** [comment]
LGTM otherwise.

**Comment by HyukjinKwon** [comment]
R and Python versions are changed relatively often, and pretty easy to change compared to Java or Scala versions .. so I agree with not mentioning as this PR proposes.

**Comment by dongjoon-hyun** [comment]
Thank you, @HyukjinKwon and @zero323 . The PR is updated!

**Comment by dongjoon-hyun** [comment]
I'll merge this. Thank you all!

**Pull request merged**